tests/unit/meta/test_snap_yaml.py

```diff
@@ -593,6 +593,62 @@ def test_project_environment_ld_library_path_null(simple_project, new_dir):
     )
 
 
+@pytest.mark.parametrize(
+    "ld_library_path",
+    [{}, {"LD_LIBRARY_PATH": None}, {"LD_LIBRARY_PATH": "test-ld-library-path"}],
+)
+@pytest.mark.parametrize("path", [{}, {"PATH": None}, {"PATH": "test-path"}])
+@pytest.mark.parametrize(
+    "other_var", [{}, {"OTHER_VAR": None}, {"OTHER_VAR": "test-foo"}]
+)
+def test_project_environment_classic_confinement(
```

Comment:
Are we testing non-classic to make sure it's still working?

Reply:
Yes, the existing tests use `strict` confinement.

Reply:
I started to refactor and parameterize the existing tests but decided not to change them to ensure I am not changing non-classic behavior.